src/RequestHandler.cpp

```diff
@@ -43,42 +44,52 @@ void RequestHandler::checkMetadataModeArguments(
     showTopicPartitionOffsets(UserArguments);
 }
 
-std::string RequestHandler::subscribeConsumeAtOffset(std::string TopicName,
-                                                     int64_t Offset) {
+void RequestHandler::subscribeConsumeAtOffset(std::string TopicName,
+                                              int64_t Offset) {
   int64_t EOFPartitionCounter = 0,
           NumberOfPartitions =
               KafkaConnection->getNumberOfTopicPartitions(TopicName);
   std::pair<std::string, bool> MessageAndEOF;
   int i = 0;
   // SUBSCRIBE AT AN OFFSET
   KafkaConnection->subscribeAtOffset(Offset, TopicName);
+  FlatbuffersTranslator FlatBuffers;
+
   while (EOFPartitionCounter < NumberOfPartitions) {
     MessageAndEOF = KafkaConnection->consumeFromOffset();
-    i++;
+    if (!MessageAndEOF.first.empty() &&
+        MessageAndEOF.first != "HiddenSecretMessageFromLovingNeutron") {
+      FlatBuffers.TakeFileID(MessageAndEOF.first);
+      i++;
+    }
     if (MessageAndEOF.second)
       EOFPartitionCounter++;
-    std::cout << MessageAndEOF.first << "\n";
+    // std::cout << MessageAndEOF.first << "\n";
   }
-  return MessageAndEOF.first;
 }
 
-std::string
-RequestHandler::subscribeConsumeNLastMessages(std::string TopicName,
-                                              int64_t NumberOfMessages) {
+void RequestHandler::subscribeConsumeNLastMessages(std::string TopicName,
+                                                   int64_t NumberOfMessages) {
   int EOFPartitionCounter = 0,
       NumberOfPartitions =
           KafkaConnection->getNumberOfTopicPartitions(TopicName);
   std::pair<std::string, bool> MessageAndEOF;
   int i = 0;
   KafkaConnection->subscribeToLastNMessages(NumberOfMessages, TopicName);
+  FlatbuffersTranslator FlatBuffers;
   while (EOFPartitionCounter < NumberOfPartitions) {
```

Comment:
Is this `while` loop the same as in another function? If so, refactor into a separate function.